### ISSUE TITLE
<a grave error>It works well in a simulator, but it doesn't work in real iPhone devices.

### ISSUE THREAD

**Issue by intelryzen** [memspace/zefyr#459]
## Steps to Reproduce

Thank you always for your hard work.

Currently, this version cannot be changed to bold text when entering Chinese or Korean. As soon as I write a letter, the thickness of the letter goes back to its original state.

This seems to be an overall problem, not just a version of this plug-in.
## Logs

```
```

```
```

**Comment by 8075qy** [comment]
Have you solved the problem now?

**Comment by intelryzen** [comment]
> Have you solved the problem now?
No, I tested it with the actual device, but the problem still remains.

**Comment by 8075qy** [comment]
After observation, I come to the conclusion that the original Pinyin will be replaced after Chinese editing, resulting in the final result that Chinese is not bold or slanted. Maybe we can change the source code to solve this problem.

**Comment by singerdmx** [comment]
@8075qy do you have a pull request to address this?

**Comment by 8075qy** [comment]
> @ 8075qy您是否有请求请求来解决此问题？

I'm modifying the source code

**Comment by singerdmx** [comment]
Could you share your solution, please?

**Comment by 8075qy** [comment]
> Could you share your solution, please?

Please leave your email and I will send the source code to your email after modification

**Comment by singerdmx** [comment]
My email is singerdmx@gmail.com
Please also leave instruction about the diff. If you have git history, that will do.

**Comment by 8075qy** [comment]
> My email is [singerdmx@gmail.com](mailto:singerdmx@gmail.com)
> Please also leave instruction about the diff. If you have git history, that will do.

What language do you have this problem in

**Comment by singerdmx** [comment]
I haven't tried what you tested with, but I do want to address this issue for my repo https://github.com/singerdmx/flutter-quill

**Comment by singerdmx** [comment]
@8075qy  I am trying to build a text editor for mobile only and fully support delta - exact the same delta as used by ReactQuill https://github.com/zenoamaro/react-quill
We already have used it in the Bullet Journal App: https://bulletjournal.us/home/index.html
We want to improve the user experience for Chinese input.
But we haven't tested using Chinese

**Comment by 8075qy** [comment]
> @ 8075qy 我正在尝试构建仅用于移动设备并完全支持增量的文本编辑器-与ReactQuill使用的完全相同的增量https://github.com/zenoamaro/react-quill
> 我们已经在Bullet Journal App中使用了它：https://bulletjournal.us/home/index.html
> 我们希望改善中文输入的用户体验。
> 但是我们还没有使用中文进行测试

Thanks, but I still want to use rich text in zerfy

**Comment by singerdmx** [comment]
Sounds good.
Please share your solution once you have it. Thanks!

**Comment by intelryzen** [comment]
thanks, koean and chinese is not well

**Comment by 8075qy** [comment]
> 谢谢，韩文和中文在此插件中效果不佳

What is your final choice

**Comment by 8075qy** [comment]
> > 谢谢，韩文和中文在此插件中效果不佳
> 
> What is your final choice

Android doesn't seem to have this problem

**Comment by intelryzen** [comment]
I just tested Korean with an Android device, but the font thickness didn't apply.
Can you test it in Korean?

**Comment by 8075qy** [comment]
> I just tested Korean with an Android device, but the font thickness didn't apply.
> Can you test it in Korean?


Android device I tested in Chinese and Korean, the results are normal

**Comment by intelryzen** [comment]
Oh? That's weird.
Did you test it with the actual device?
When I just tested it, it didn't work out. I'll double-check.

**Comment by 8075qy** [comment]
> Oh? That's weird.
> Did you test it with the actual device?
> When I just tested it, it didn't work out. I'll double-check.

Yes, I use the physical machine for testing

**Comment by intelryzen** [comment]
Could I know the zefyr version that you used?

**Comment by 8075qy** [comment]
> Could I know the zefyr version that you used?

0.12.0

**Comment by 8075qy** [comment]
在zefyr依赖包notus中有个insert_rules.dart中修改规则,因为每个国家语言不一致导致键盘的输出不一样,按照自己国家的语言修改就行

**Comment by 8075qy** [comment]
There is an insert in the zefyr dependency package notus_ rules.dart Because the language of each country is different, the output of the keyboard is different. Just modify it according to the language of your country

**Comment by 8075qy** [comment]
There is a inser_rules.dart modification rule in the zefyr dependency package notus, because the output of the keyboard is different due to the different languages of each country. Just modify it according to the language of your own country

**Comment by intelryzen** [comment]
Unlike English, Korean has up to three keyboard characters in one box. I knew that the thick letters would disappear as soon as there were several characters in one box. I saw insert_rule.dart, but I don't know which code to modify

**Comment by 8075qy** [comment]
> Unlike English, Korean can have up to three keyboard characters in one box. I knew that the thick letters would disappear as soon as there were several characters in one box. I saw insert_rule.dart, but I don't know which code to modify

In fact, the most important problem is the loss of style state in controller.dart. You can go into debugging mode and look at the toggledStyles function. When the word has not been generated, the style can be obtained. After the word is determined, the style cannot be obtained

**Comment by 8075qy** [comment]
_toggledStyles = NotusStyle();  After reset, there is no style after confirming the word

**Comment by 8075qy** [comment]
https://user-images.githubusercontent.com/58923624/104315032-4c6fc680-5515-11eb-85b1-3af12963db66.MOV

iphone

**Comment by 8075qy** [comment]
![Uploading 85B6201A-8519-4FD9-A276-E21D531BCB74.gif…]()
iphone

**Comment by 8075qy** [comment]
![IMG_0533](https://user-images.githubusercontent.com/58923624/104349168-88b71d00-553d-11eb-9232-1cfea2c60360.GIF)
Iphone test results.

**Comment by intelryzen** [comment]
Wow, that's amazing. Could you send me the modified source code by email below?
intelryzen@naver.com
